diff --git a/entrypoint.sh b/entrypoint.sh
index 778c64e..4d12aeb 100755
--- a/entrypoint.sh
+++ b/entrypoint.sh
@@ -5,7 +5,7 @@
 # https://hub.docker.com/r/erichough/nfs-server
 # https://github.com/ehough/docker-nfs-server
 #
-# Copyright (C) 2017-2019  Eric D. Hough
+# Copyright (C) 2017-2020  Eric D. Hough
 #
 # This program is free software: you can redistribute it and/or modify
 # it under the terms of the GNU General Public License as published by
@@ -75,16 +75,6 @@ readonly STATE_NFS_VERSION='nfs_version'
 declare -A state


-######################################################################################
-### string utils
-######################################################################################
-
-toupper() {
-
-  echo "$1" | awk '{ print toupper($0) }'
-}
-
-
 ######################################################################################
 ### logging
 ######################################################################################
@@ -110,7 +100,7 @@ log_header() {

   echo "
 ==================================================================
-      $(toupper "$1")
+      ${1^^}
 =================================================================="
 }

@@ -283,7 +273,7 @@ is_kernel_module_loaded() {

 is_granted_linux_capability() {

-  if capsh --print | grep -Eq "^Current: = .*,?${1}(,|$)"; then
+  if capsh --print | grep -Eq "^Current: ?=? .*,?${1}(,|$)"; then
     return 0
   fi

@@ -341,7 +331,8 @@ assert_port() {
 init_state_logging() {

   # if the user didn't request a specific log level, the default is INFO
-  local -r normalized_log_level=$(toupper "${!ENV_VAR_NFS_LOG_LEVEL:-$LOG_LEVEL_INFO}")
+  local incoming_log_level="${!ENV_VAR_NFS_LOG_LEVEL:-$LOG_LEVEL_INFO}"
+  local -r normalized_log_level="${incoming_log_level^^}"

   if ! echo "$normalized_log_level" | grep -Eq 'DEBUG|INFO'; then
     bail "the only acceptable values for $ENV_VAR_NFS_LOG_LEVEL are: DEBUG, INFO"
